pkg/image/vulnerability_scan.go

```diff
@@ -0,0 +1,118 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+package image
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"sort"
+	"strconv"
+	"strings"
+
+	buildapi "github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+)
+
+const (
+	VulnerabilityCountLimit = 50 // Number of vulnerabilities to be added to buildrun output.
+)
+
+func RunVulnerabilityScan(ctx context.Context, imagePath string, settings buildapi.VulnerabilityScanOptions) ([]buildapi.Vulnerability, error) {
+
+	trivyArgs := []string{"image", "--quiet", "--format", "json", "--input", imagePath}
+
+	if settings.IgnoreOptions != nil {
+		if settings.IgnoreOptions.Severity != "" {
+			severity := getSeverityStringForTrivyScan(settings.IgnoreOptions.Severity)
+			trivyArgs = append(trivyArgs, "--severity", severity)
+		}
+		if len(settings.IgnoreOptions.Issues) > 0 {
+			// Create a file with vulnerabilities to be ignored.
+			ignoreFile, err := os.CreateTemp("", "ignore")
+			if err != nil {
+				log.Printf("Could not create file for ignored vulnerabilities: %v\n", err)
+				return nil, err
```

Comment:
Would be interested in other opinions as well, but I kind of think that depending where this is used, it is more universal to have the error wrapped in context rather than "just" a log line. What do you think?
```suggestion
				return nil, fmt.Errorf("could not create file for ignored vulnerabilities: %w", err)
```

Reply:
I personally would have implemented it without writing an ignore file, but would have filtered the result myself. Beside that, I tend to also return the error and then on top-level, the main function can print the error no matter from where it comes.

Reply:
Done.